src/uhid/joypad_ps5.cpp

```
@@ -265,10 +258,24 @@ Result<PS5Joypad> PS5Joypad::create(const DeviceDefinition &device) {
     def.report_description = {&uhid::ps5_rdesc[0], &uhid::ps5_rdesc[0] + sizeof(uhid::ps5_rdesc)};
   }
 
-  auto joypad = PS5Joypad(device.vendor_id);
+  std::array<unsigned char, 6> mac_address = {};
+  if (def.uniq.empty()) {
```

I think this part would be better written with the following as it really checks if the MAC is in the proper format:
```c++
if (!def.uniq.empty() && std::regex_match(def.uniq, (std::regex) "^([0-9A-Fa-f]{2}[:]){5}([0-9A-Fa-f]{2})$")) {
    // Assuming we have in input a MAC address in the format of xx:xx:xx:xx:xx:xx
    std::stringstream ss(def.uniq);
    for (int i = 0; i < 6; ++i) {
      unsigned int value;
      ss >> std::hex >> value;
      mac_address[i] = static_cast<unsigned char>(value);
      if (i < 5)
        ss.ignore(1, ':');
    }
  } else {
    mac_address = generate_mac_address();
  }
```
This needs ```#include <regex>``` but makes the whole section a lot safer and falls back to a random mac if it is not present or invalid.

---

Other than that this looks good to me. Also tested and is working for me.

---

I see what you mean, but I feel that it's out of scope for inputtino.  
This is a fairly low level library after all; there are tons of way to break things if someone really wants to. This kind of checks should be done upstream, where you have proper logging, exception handling and so on. Besides, none of this should be directly exposed to user input..

---

Since the fix seems to work for you too I'm going to merge this so that you can update the git submodule to the latest `stable` branch.

---

All good, thanks!